### DISCUSSION TITLE
Viessmann PT2

### DISCUSSION THREAD

**Discussion by creiser-jambit** [open3e/open3e#137] [General]
Hi,
ich habe hier eine Vissmann PT2 und open3e_depictSystem  war erfolgreich. Ich habe 607 founds bekommen. Genial.
In den Open3Edatapoints_XXX (3 Stück) sind aber ganz viele "none" zu lesen. Kann mir sagen, was hier noch passieren muss?
[logfile.txt](https://github.com/user-attachments/files/17482195/logfile.txt)
[open3e_config.tgz](https://github.com/user-attachments/files/17482200/open3e_config.tgz)

Der für mich interessanteste Wert ist die aktuelle Stromerzeugung und danach der aktuelle Haushaltsverbrauch. Was gefunden wurde:
found 567:24:GeneratedElectricity
found 1214:2:ElectricalPowerOutput

Hat schon jemand die Werte vom Smartmeter irgendwo gefunden?

ps: Aktuell habe ich noch keinen CAN Adapter und kein MQTT.

**Comment by philippoo66** [comment]
moin @creiser-jambit !

das mit dem 'none' ist normal. die Open3Edatapoints_XXX funktionieren nach den 'Overlay-Prinzip': 
- DIDs, die nicht auftauchen, existieren nicht auf dem Gerät, 
- da wo 'none' steht ist keine Änderung (der Datenlänge) gegenüber der 'generischen' Open3eDatapoints, 
- da wo mehr steht (RawCodec) ist die Datenlänge anders als in der generischen Liste (oder sie stehen noch nicht in der generischen Liste), und _da_ sollte man dann mal näher schauen, dass man die Codierung geklärt bekommt.

Die Energiezähler sind Zukaufprodukte und tauchen von daher nicht in dem Viessmann UDS System auf, sondern melden sich mit CAN Frames ohne höheres Protokoll. @MyHomeMyData Jürgens https://github.com/MyHomeMyData/E3onCAN 'lauscht' auch diesen Werten und stellt sie auf MQTT. Das kannst du neben open3e nebenher laufen lassen, solltest dann vlt ein paar Sachen der PT2 rausnehmen, damit das nicht doppelt ist.

Falls du ioBroker nutzt, solltest du Jürgens ioBroker-Adapter benutzen, der vereint beides in einer sehr benutzerfreundlichen Weise. Ich denke, er wird sich in Kürze melden und dir 'auf's Pferd helfen'.

Grüsse!
Phil

**Reply by MyHomeMyData** [reply]
@creiser-jambit Moin! Phil hat Dir ja schon alle wesentlichen Infos gegeben. Falls Du den [ioBroker-Adapter](https://github.com/MyHomeMyData/ioBroker.e3oncan#readme) verwenden möchtest, einfach in der Adapterübersicht nach "e3oncan" suchen. open3e ist dann nicht mehr notwendig. Der Adapter unterstützt die Energy Meter E380CA und E3100CB.

Falls Du Fragen hast, gerne melden.

Viele Grüße
Jürgen

**Reply by creiser-jambit** [reply]
Erst einmal danke für die Antwort. E3onCAN werde ich mir mal´ ansehen. Zwei Fragen/Anmerkungen habe ich tatsächlich:
* Ich habe einen anderen Smart Meter verbaut - 
![PXL_20231129_205233363](https://github.com/user-attachments/assets/fe88903a-c4ac-4734-ad03-d3badcb2026d)
* Gibt es ein Foto, wo/wie der CAN bei der PT-2 angeschlossen ist? Zum Testen werde ich auf dem Wifi HotSpot bleiben, aber langfristig wollte ich dann auf CAN umsteigen. Noch habe ich nur die obere Klappe geöffnet und noch nicht weiter die Heizung zerlegt ;-)

**Reply by MyHomeMyData** [reply]
Soweit ich es verstanden habe, wird bei der PT2 der Energiezähler über M-Bus angeschlossen. Ob Dein Smart-Meter (auch) einen CAN-Bus hat, sollte aus der Betriebsanleitung hervorgehen. Wahrscheinlich wird aber nur das verfügbar sein, was die PT2 über CAN oder AP zur Verfügung stellt.

Der CAN-Bus der PT2 ist wohl über Klemme 91 zugänglich. Evtl. musst Du den Abschlusswiderstand über den Schalter unter den Klemmen abschalten und am USB2CAN-Adapter aktivieren. Das hängt von der weiteren Verkabelung am CAN-Bus ab.

**Reply by creiser-jambit** [reply]
Okay. Dann muss ich bei Gelegenheit mal´ meine Heizung aufschrauben. Im Handbuch zum Smart Meter finde ich nichts zu CAN - schade. https://sbc-support.com/uploads/tx_srcproducts/31-024_GER_DS_Energiezaehler-ALE3-BiDi-mit-M-Bus_01.pdf
